addons/GME/Scripts/Game/GME/Systems/Persistence/Serializers/GME_HitZoneContainerComponentSerializer.c

```diff
@@ -0,0 +1,33 @@
+//------------------------------------------------------------------------------------------------
+class GME_HitZoneContainerComponentSerializer : HitZoneContainerComponentSerializer
+{
+	//------------------------------------------------------------------------------------------------
+	override protected ESerializeResult Serialize(notnull IEntity owner, notnull GenericComponent component, notnull BaseSerializationSaveContext context)
+	{
+		ESerializeResult result = super.Serialize(owner, component, context);
+		if (result == ESerializeResult.ERROR)
+			return result;
+		
+		SCR_DamageManagerComponent damageManager = SCR_DamageManagerComponent.Cast(component);
+		if (!damageManager.IsDamageHandlingEnabled())
+		{
+			context.WriteValue("damageHandlingEnabled", false);
+			result = ESerializeResult.OK;
+		}
+		
+		return result;
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	override protected bool Deserialize(notnull IEntity owner, notnull GenericComponent component, notnull BaseSerializationLoadContext context)
+	{
+		super.Deserialize(owner, component, context);
+		SCR_DamageManagerComponent damageManager = SCR_DamageManagerComponent.Cast(component);
+		
+		bool damageHandlingEnabled;
+		if (context.Read(damageHandlingEnabled))
+			damageManager.EnableDamageHandling(false);
```

Comment:
Not sure I follow this flow precisely. We initiate ``damageHandlingEnabled`` and then use ``context.Read()``. Does this read the value into the ``damageHandlingEnabled`` variable, or is it only passed to the function for matching the name?

if the ``read()`` returns the bool value saved, we then say ``if damageHandlingEnabled == true`` then we disable damage? Is that not the wrong way around?

We are also assuming here that by default when this is run, damage-handling on the entity will be turned on. Is that a fair assumption? Can we have an instance where we have saved damage-handling as "on", but something else in the loading process has turned it off. So we would end up with it being off here, even if our persistence data said it should be on?

Reply:
Just changed it accordingly.